### ISSUE TITLE
Hidden cookie logic confuses a screenreader somewhat

### ISSUE THREAD

**Issue by m--** [posit-dev/product-doc-theme#102]
## Describe the bug
The cookie icon is not accessible via VoiceOver [when hidden](https://github.com/posit-dev/product-doc-theme/blob/main/_extensions/posit-docs/assets/_analytics.html#L18), which is great. However, since the logic that hides the cookie icon acts on the `<a>` that contains the icon and not on the containing `<li>` element, VoiceOver announces the list as having "5 items" when, in fact, only 4 are visible/accessible. We should tweak [the logic that determines what to hide](https://github.com/posit-dev/product-doc-theme/blob/main/_extensions/posit-docs/assets/_analytics.html#L13) to act instead on the containing `<li>`, which would make the "list of items" count be accurate.

### To reproduce
1. `quarto preview` any docs site using v7 of the theme
2. assuming you have a Mac, start VoiceOver
3. navigate to the footer
4. VoiceOver announces a list of "5 items" when in fact there are only four. (Or "four items" when there are only 3)

### Expected behavior
VoiceOver should announce the length of the list accurately
